contrib/intel/jenkins/tests.py

```diff
@@ -762,3 +762,92 @@ def execute_cmd(self, oneccl_test_gpu):
+                 'rpc_two_node'              :       {'tags' :'cart,rpc,two_node','numservers':2, 'numclients':1},
+                 'swim_notification'         :       {'tags' :'cart,rpc,swim_rank_eviction,one_node', 'numservers':1, 'numclients':1}
+        }
+
+
+    def set_paths(self):
+        self.ci_middlewares_path = f'{ci_site_config.ci_middlewares}'
+        self.daos_install_root = f'{self.ci_middlewares_path}/daos/install'
+        self.cart_test_scripts = f'{self.daos_install_root}/lib/daos/TESTING/ftest'
+        self.mpipath = f'{ci_site_config.daos_mpi}/bin'
+        self.pathlist = [f'{self.daos_install_root}/bin/', self.cart_test_scripts, self.mpipath, \
+                       f'{self.daos_install_root}/lib/daos/TESTING/tests']
+        self.daos_prereq = f'{self.daos_install_root}/prereq'
+        common.run_command(['rm','-rf', f'{self.daos_prereq}/debug/ofi'])
+        common.run_command(['ln', '-sfn', self.libfab_installpath, f'{self.daos_prereq}/debug/ofi'])
+
+    def set_environment(self, core_prov, util_prov):
+        if (core_prov == 'verbs'):
+            os.environ["OFI_DOMAIN"] = 'mlx5_0'
+        else:
+            os.environ["OFI_DOMAIN"] = 'ib0'
+        os.environ["OFI_INTERFACE"] = 'ib0'
+        os.environ["CRT_PHY_ADDR_STR"] = f'ofi+{core_prov};ofi_{util_prov}'
+        os.environ["PATH"] += os.pathsep + os.pathsep.join(self.pathlist)
+        os.environ["DAOS_TEST_SHARED_DIR"] = ci_site_config.daos_share
+        os.environ["DAOS_TEST_LOG_DIR"] = ci_site_config.daos_logs
+        os.environ["LD_LIBRARY_PATH"] = f'{self.ci_middlewares_path}/daos/install/lib64:{self.mpipath}'
+
+    @property
+    def cmd(self):
+        return "./launch.py "
+
+    def options(self, testname):
+        opts = "-s "
+        opts += f"{self.cart_tests[testname]['tags']} "
+
+        if (self.cart_tests[testname]['numservers'] != 0):
+            servers = ",".join(self.daos_nodes[:self.cart_tests[testname]['numservers']])
+            opts += f"--test_servers={servers} "
+        if (self.cart_tests[testname]['numclients'] != 0):
+            clients = ",".join(self.daos_nodes[:self.cart_tests[testname]['numclients']])
+            opts += f"--test_clients={clients}"
+        return opts
+
+    @property
+    def execute_condn(self):
+        return True
+    def execute_cmd(self):
+        sys.path.append(f'{self.daos_install_root}/lib64/python3.6/site-packages')
+        os.environ['PYTHONPATH']=f'{self.daos_install_root}/lib64/python3.6/site-packages'
+        print("PATH:" +  os.environ["PATH"])
+        print("LD_LIBRARY_PATH:" + os.environ["LD_LIBRARY_PATH"])
+        print("MODULEPATH:" +  os.environ["MODULEPATH"])
+
+        test_dir=self.cart_test_scripts
+        curdir=os.getcwd()
+        os.chdir(test_dir)
+        for test in self.cart_tests:
+            print(test)
+            command = self.cmd + self.options(test)
+            outputcmd = shlex.split(command)
+            common.run_command(outputcmd)
```

Comment:
Can you double check that everything you previously set did get set? I am not convinced all of the environment variables you set before calling common.run_command are actually saved for the process that is running your test.

Reply:
The test script (launch.py) prints out the test related environment variables. These can be seen in the Jenkins log. The test would fail to run properly if the PATH and LD_LIBRARY_PATH were not set. 

16:13:04  ./launch.py -s cart,corpc,one_node --test_servers=d3-ib0
16:13:04  None
16:13:04  Arguments: Namespace(archive=False, clean=False, disable_stop_daos=False, discard=False, extra_yaml=None, failfast=False, include_localhost=False, insecure_mode=False, jenkinslog=False, list=False, logs_threshold=None, mode='normal', modify=False, nvme=None, process_cores=False, rename=False, repeat=1, sparse=True, tags=['cart,corpc,one_node'], test_clients=None, test_servers='d3-ib0', timeout_multiplier=None, verbose=0, yaml_directory=None)
16:13:04  interf is ib0
16:13:04  Testing with ib0 as the default interface
16:13:04  Testing with OFI_INTERFACE=ib0
16:13:04  Testing with DAOS_TEST_FABRIC_IFACE=ib0
16:13:04  Testing with CRT_CTX_SHARE_ADDR=0
16:13:04  Testing with CRT_PHY_ADDR_STR=ofi+verbs;ofi_rxm
16:13:04  Testing with PYTHONPATH=/home/cstbuild/ci_middlewares/daos/install/lib64/python3.6/site-packages:/home/cstbuild/ci_middlewares/daos/install/lib/daos/TESTING/ftest/util/apricot:/home/cstbuild/ci_middlewares/daos/install/lib/daos/TESTING/ftest/util:/home/cstbuild/ci_middlewares/daos/install/lib/daos/TESTING/ftest/cart:/usr/local/lib64/python3.6/site-packages:/usr/local/lib/python3.6/site-packages:/usr/lib64/python3.6/site-packages:/usr/lib/python3.6/site-packages

Reply:
Perfect, thank you